### PR TITLE
fixed client input to reflect prev choice

### PR DESCRIPTION
**Pull request by Arjunica**
UN-tested. Can someone test that the branch does not break any thing

### CODE REVIEW

**Comment by QuinnWass** [comment]
Ill review this afternoon. lgtm at first glance

**Comment by Arjunica** [comment]
Screen Shot 2020-10-25 at 11 21 28 AM
Looks like the arg is coming back null. I am not sure why tho. are we freeing it somewhere else?

**Comment by Arjunica** [comment]
Screen Shot 2020-10-25 at 11 25 25 AM

I think I am just missing something. I'm gonna look at the code again

**Comment by Arjunica** [comment]
nvm I see it

File: README.md [hunk 1]
```diff
@@ -76,3 +76,130 @@ Receives long-lat cords from the places server and then uses a KD tree with knn
...
   + Pro: efficiently insert and retrival 
   + Con: memory may not be efficiently be managed as this is an STL container.
+
+- Trie:
+  + Pro: efficiently do a prefix search
+  + Con: lots of memory for storing the strings
+
+- STL Map:
+  + Pro: efficiently insert and retrival 
+  + Con: memory may not be efficiently be managed as this is an STL container.
```
Comment (galaxyonfire) [inline README.md:86]: is this not a copy of Trie/STL entries above?

Comment (Arjunica) [reply README.md:86]: I wrote them separately but they are similar 🤔 feel free to edit...

**Comment by galaxyonfire** [comment]
Also seems that when you enter a correct state after incorrect state input, it does not carry over
(original input was `./places_client localhost no XX`)
image
Note sure if this is expected result or not

File: README.md (hunk 1 above)
Comment (galaxyonfire) [reply README.md:86]:
```suggestion
```

**Comment by Arjunica** [comment]
> Also seems that when you enter a correct state after incorrect state input, it does not carry over
> (original input was `./places_client localhost no XX`)
> image
> Note sure if this is expected result or not

if correct state we do not "re-poll" for state... its an easy change tho if you want to do it

**Comment by galaxyonfire** [comment]
Seems there is a bug where the incorrect state is printed after putting in correct name/state:
image

**Comment by galaxyonfire** [comment]
seems state is never replaced with s_state before line 87

File: src/places_client.c [hunk 2]
```diff
@@ -78,6 +72,12 @@ places_server_1(char *host, char *city, char *state)
 		}
```
Comment (galaxyonfire) [inline src/places_client.c:72]:
```suggestion
		}
	if (result_1->status == 0)
	        state = strdup(s_state.c_str());
```

Comment (galaxyonfire) [reply src/places_client.c:72]: @Arjunica ?

Comment (Arjunica) [reply src/places_client.c:72]: Ya I see the issue. I would say we update state and city param at the same time we set them into find_closest_airport_given_name_1_arg

Comment (galaxyonfire) [reply src/places_client.c:72]: don't think we need to update the city param b/c the true city name is pulled from the return value.

Comment (galaxyonfire) [reply src/places_client.c:72]: we only need to update the state value once (when we know the search was completed succesfully).

Comment (Arjunica) [reply src/places_client.c:72]: sounds good. Feel free to make the edit.

Comment (galaxyonfire) [reply src/places_client.c:72]: tested on cs1, works as expected with no mem leaks

**Review by galaxyonfire — APPROVED** [review]
looks gud 2 me

**Pull request merged**